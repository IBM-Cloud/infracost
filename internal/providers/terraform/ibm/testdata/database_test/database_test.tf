terraform {
  required_providers {
    ibm = {
      source  = "IBM-Cloud/ibm"
      version = "1.63.0"
    }
  }
}

provider "ibm" {
  region = "us-south"
}

resource "ibm_database" "test_db1" {
  name     = "demo-postgres"
  service  = "databases-for-postgresql"
  plan     = "standard"
  location = "eu-gb"

  group {
    group_id = "member"
    memory {
      allocation_mb = 12288
    }
    disk {
      allocation_mb = 131072
    }
    cpu {
      allocation_count = 3
    }
  }
  configuration = <<CONFIGURATION
  {
    "max_connections": 400
  }
  CONFIGURATION
}

resource "ibm_database" "test_db2" {
  name     = "demo-postgres2"
  service  = "databases-for-postgresql"
  plan     = "standard"
  location = "eu-gb"

  group {
    group_id = "member"
    memory {
      allocation_mb = 15360
    }
    members {
      allocation_count = 4
    }
  }
  configuration = <<CONFIGURATION
  {
    "max_connections": 400
  }
  CONFIGURATION
}

resource "ibm_database" "test_es_enterprise_db1" {
  name                         = "demo-es-enterprise"
  service                      = "databases-for-elasticsearch"
  plan                         = "enterprise"
  location                     = "eu-gb"

  group {
    group_id = "member"
    memory {
      allocation_mb = 12288
    }
    disk {
      allocation_mb = 131072
    }
    cpu {
      allocation_count = 3
    }
  }
}

resource "ibm_database" "test_es_enterprise_db2" {
  name                         = "demo-es-enterprise2"
  service                      = "databases-for-elasticsearch"
  plan                         = "enterprise"
  location                     = "eu-gb"

  group {
    group_id = "member"
    members {
      allocation_count = 4
    }
    memory {
      allocation_mb = 15360
    }
  }
}

# resource "ibm_database" "test_es_platinum_db1" {
#   name                         = "demo-es-platinum"
#   service                      = "databases-for-elasticsearch"
#   plan                         = "platinum"
#   location                     = "eu-gb"

#   group {
#     group_id = "member"
#     memory {
#       allocation_mb = 12288
#     }
#     disk {
#       allocation_mb = 131072
#     }
#     cpu {
#       allocation_count = 3
#     }
#   }
# }

# resource "ibm_database" "test_es_platinum_db2" {
#   name                         = "demo-es-platinum"
#   service                      = "databases-for-elasticsearch"
#   plan                         = "platinum"
#   location                     = "eu-gb"

#   group {
#     group_id = "member"
#     members {
#       allocation_count = 3
#     }
#     memory {
#       allocation_mb = 1024
#     }
#   }
# }