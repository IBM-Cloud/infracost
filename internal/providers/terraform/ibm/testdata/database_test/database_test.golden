
 Name                                      Monthly Qty  Unit                    Monthly Cost 
                                                                                             
 ibm_database.test_db1                                                                       
 ├─ RAM (first 1 GB-RAM)                             1  GB-RAM                         $5.23 
 ├─ RAM (over 0 GB-RAM)                              1  GB-RAM                         $5.39 
 ├─ Disk (first 1 GB-DISK)                           1  GB-DISK                        $0.61 
 ├─ Disk (over 0 GB-DISK)                            1  GB-DISK                        $0.63 
 └─ Core (first 1 Virtual Processor Core)            1  Virtual Processor Core        $31.40 
 └─ Core (over 0 Virtual Processor Core)             1  Virtual Processor Core        $32.34 
                                                                                             
 ibm_database.test_db2                                                                       
 ├─ RAM (first 1 GB-RAM)                             1  GB-RAM                         $5.23 
 ├─ RAM (over 0 GB-RAM)                              1  GB-RAM                         $5.39 
 └─ Disk (first 1 GB-DISK)                           1  GB-DISK                        $0.61 
 └─ Disk (over 0 GB-DISK)                            1  GB-DISK                        $0.63 
                                                                                             
 ibm_database.test_es_enterprise_db1                                                         
 ├─ RAM (first 1 GB-RAM)                             1  GB-RAM                        $15.70 
 ├─ RAM (over 0 GB-RAM)                              1  GB-RAM                        $16.17 
 ├─ Disk (first 1 GB-DISK)                           1  GB-DISK                        $0.61 
 ├─ Disk (over 0 GB-DISK)                            1  GB-DISK                        $0.63 
 └─ Core (first 1 Virtual Processor Core)            1  Virtual Processor Core        $31.40 
 └─ Core (over 0 Virtual Processor Core)             1  Virtual Processor Core        $32.34 
                                                                                             
 ibm_database.test_es_enterprise_db2                                                         
 ├─ RAM (first 1 GB-RAM)                             1  GB-RAM                        $15.70 
 ├─ RAM (over 0 GB-RAM)                              1  GB-RAM                        $16.17 
 └─ Disk (first 1 GB-DISK)                           1  GB-DISK                        $0.61 
 └─ Disk (over 0 GB-DISK)                            1  GB-DISK                        $0.63 
                                                                                             
 ibm_database.test_es_platinum_db1                                                           
 ├─ RAM                                             36  GB-RAM                       $939.60 
 ├─ Disk                                           384  GB-DISK                      $234.24 
 └─ Core                                             9  Virtual Processor Core       $282.60 
                                                                                             
 ibm_database.test_es_platinum_db2                                                           
 ├─ RAM                                              4  GB-RAM                       $104.40 
 └─ Disk                                            20  GB-DISK                       $12.20 
                                                                                            
 OVERALL TOTAL                                                                     $1,790.45 
──────────────────────────────────
4 cloud resources were detected:
∙ 4 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file